node/_bin/munin-asyncd.in

```diff
@@ -143,7 +143,11 @@ my $last_cleanup=0;
 MAIN: while($keepgoing) {
 	my $when = time;
 
-	my $when_next = $when + $timeout; # wake up at least every $timeout sec
+	my $when_next = int($when / $minrate) * $minrate + 180;
+	if ($when_next <= $when) {
+		$when_next = $when_next + $minrate;
```

Comment:
What do you want to achieve with this conditional?
Is it sufficient to add `$minrate` only once, even if the condition may still be true? Or would `while` be more suitable than `if`?

Reply:
As discussed in IRC 2018-09-21, removing this "if" causes munin-asyncd run in busyloop during update, opening SPOOL-META again and again, over 200000 times, eating 20s of cpu time. By adding debug there is see $when_next being equal or below $when during update. Simple "if" seems to be enough, as lowest value was "$when - 60", but "while" is safer.